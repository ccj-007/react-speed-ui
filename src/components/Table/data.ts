export const defaultDataSource = [
	{
		key: "1",
		name: "乌龟",
		age: 32,
		address: "某个水域",
	},
	{
		key: "2",
		name: "兔子",
		age: 42,
		address: "某个公园",
	},
	{
		key: "3",
		name: "老鹰",
		age: 22,
		address: "某个天空",
	},
	{
		key: "4",
		name: "乌龟22",
		age: 32,
		address: "某个水域22",
	},
	{
		key: "5",
		name: "兔子22",
		age: 42,
		address: "某个公园22",
	},
	{
		key: "6",
		name: "老鹰22",
		age: 22,
		address: "某个天空22",
	},
];
