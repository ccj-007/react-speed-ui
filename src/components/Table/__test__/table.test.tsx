import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Table } from '../../index';
import { defaultDataSource } from '../data';

const paginationParams = {
  total: 6,
  defaultCurrent: 1,
  current: 1,
  defaultPageSize: 3,
  pageSize: 3,
};
const columns = [
  {
    title: 'ID',
    dataIndex: 'key',
    key: 'key',
  },
  {
    title: '姓名',
    dataIndex: 'name',
    key: 'name',
  },
  {
    title: '年龄',
    dataIndex: 'age',
    key: 'age',
  },
  {
    title: '住址',
    dataIndex: 'address',
    key: 'address',
  },
];

describe('test Table component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Table dataSource={defaultDataSource} columns={columns} paginationParams={paginationParams} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
