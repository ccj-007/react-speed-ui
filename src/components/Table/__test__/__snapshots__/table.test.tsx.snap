// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Table component should render 1`] = `
<DocumentFragment>
  <div
    class="speed-table"
  >
    <div
      class="speed-table-header"
    >
      <div
        class="speed-table-header-item"
      >
        ID
      </div>
      <div
        class="speed-table-header-item"
      >
        姓名
      </div>
      <div
        class="speed-table-header-item"
      >
        年龄
      </div>
      <div
        class="speed-table-header-item"
      >
        住址
      </div>
    </div>
    <div
      class="speed-table-body"
    >
      <div
        class="speed-table-body-row"
      >
        <div
          class="speed-table-body-row-item"
        >
          4
        </div>
        <div
          class="speed-table-body-row-item"
        >
          乌龟22
        </div>
        <div
          class="speed-table-body-row-item"
        >
          32
        </div>
        <div
          class="speed-table-body-row-item"
        >
          某个水域22
        </div>
      </div>
      <div
        class="speed-table-body-row"
      >
        <div
          class="speed-table-body-row-item"
        >
          5
        </div>
        <div
          class="speed-table-body-row-item"
        >
          兔子22
        </div>
        <div
          class="speed-table-body-row-item"
        >
          42
        </div>
        <div
          class="speed-table-body-row-item"
        >
          某个公园22
        </div>
      </div>
      <div
        class="speed-table-body-row"
      >
        <div
          class="speed-table-body-row-item"
        >
          6
        </div>
        <div
          class="speed-table-body-row-item"
        >
          老鹰22
        </div>
        <div
          class="speed-table-body-row-item"
        >
          22
        </div>
        <div
          class="speed-table-body-row-item"
        >
          某个天空22
        </div>
      </div>
    </div>
    <div
      class="speed-table-page"
    >
      <div
        class="speed-pagination"
      >
        <div
          class="speed-pagination-page"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-left fa-1x speed-icon"
            color="#000"
            data-icon="angle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              fill="currentColor"
            />
          </svg>
        </div>
        <button
          class="speed-pagination-page"
        >
          1
        </button>
        <button
          class="speed-pagination-page speed-pagination-page-active"
        >
          2
        </button>
        <div
          class="speed-pagination-page"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-right fa-1x speed-icon"
            color="#000"
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
